imagery/i.gensig/testsuite/test_i_gensig.py

```diff
@@ -0,0 +1,160 @@
+"""
+Name:      i.gensig general functionality
+Purpose:   Test ability to generate signature files
+
+Author:    Maris Nartiss
+Copyright: (C) 2022 by Maris Nartiss and the GRASS Development Team
+Licence:   This program is free software under the GNU General Public
+           License (>=v2). Read the file COPYING that comes with GRASS
+           for details.
+"""
+
+import os
+import stat
+import ctypes
+import pathlib
+import shutil
+
+from grass.pygrass import utils
+from grass.pygrass.gis import Mapset
+from grass.script.core import tempname
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+from grass.lib.gis import G_mapset_path
+from grass.lib.raster import Rast_write_semantic_label
+from grass.lib.imagery import (
+    Signature,
+    Ref,
+    I_init_group_ref,
+    I_add_file_to_group_ref,
+    I_put_group_ref,
+    I_put_subgroup_ref,
+    I_fopen_signature_file_old,
+    I_read_signatures,
+)
+
+
+class SuccessTest(TestCase):
+    """Test successful generation of a sig file"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Ensures expected computational region and generated data"""
+        cls.use_temp_region()
+        cls.runModule("g.region", n=5, s=0, e=5, w=0, res=1)
+        cls.data_dir = os.path.join(pathlib.Path(__file__).parent.absolute(), "data")
```

Comment:
This should be relative path (to the current dir), so simple `"data"` is sufficient. The [documentation](https://grass.osgeo.org/grass83/manuals/libpython/gunittest_testing.html#data-specific-to-one-test) says:

> Data specific to one test
>
> If the data required by the test are not part of standard location and cannot be part of the test file itself, this data should be stored in files in data subdirectory of testsuite directory. The test should access the data using a relative path from its location, i.e. all data will be accessed using `data/...`. This data directory might be used directly when running test file directly in the directory in the source code or might be copied to the test current working directory when running tests by the main test invoking tool.

This is also similar to how some pytest plugins work, so I think it is unlikely we will change that.